crates/hir-ty/src/method_resolution.rs

```diff
@@ -625,17 +625,22 @@ pub(crate) fn iterate_method_candidates<T>(
 }
 
 pub fn lookup_impl_method(
-    self_ty: &Ty,
     db: &dyn HirDatabase,
     env: Arc<TraitEnvironment>,
     trait_: TraitId,
     name: &Name,
+    fn_subst: Substitution,
```

Comment:
kind of weird to be passing the fn subst, but not the actual fn ID, right? we could just pass the trait subst, and then merge that with the `trait_` parameter into a `TraitRef`

Reply:
Yeah I feel weird too. This function is called from hir and I didn't want to let hir manipulate `Substitution`s (besides passing them around as-is). It's just my preference so can we just make new `Substitution` out of fn subst in hir layer?

Reply:
Hmm I think ideally we'd have some nice functions to get the trait ref from a function subst and so on. I think for now either would be fine.